File: apps/web/petProject/src/types/player.ts
import type {Pet} from "./pet";
import type {Inventory} from "./inventory";

interface dbPlayer {
    user_id: string;
    username: string;
    last_update: Date
    currency: number
}

function createDbPlayer(overrides: Partial<dbPlayer> = {}): dbPlayer {
    return {
        user_id: "test",
        username: "Test Player",
        currency: 1000,
        last_update: new Date(),
        ...overrides
    };
}

function toPlayer(player: dbPlayer): Player {
    return {
        id: player.user_id,
        username: player.username,
        lastUpdate: player.last_update,
        currency: player.currency,
        pet: null,
        inventory: null
    };
}

interface Player {
    id: string;
    username: string;
    pet: Pet | null;
    inventory: Inventory | null;
    currency: number;
    lastUpdate: Date;
}

function toDbPlayer(player: Player){
    return {
        user_id: player.id,
        username: player.username,
        last_update: new Date(),
        currency: player.currency
    }
}

function createPlayer(overrides: Partial<Player> = {}): Player {
    return {
        id: "",
        username: "Test Player",
        pet: null,
        inventory: null,
        currency: 1000,
        lastUpdate: new Date(),
        ...overrides
    };
}

export type { Player, dbPlayer};
export { createPlayer, toPlayer, toDbPlayer, createDbPlayer};

File: apps/petProject/shared/Models/Player.swift
//
//  Player.swift
//  petProject
//
//  Created by Aaron Foster on 3/10/26.
//

import Foundation
import Combine

struct dbPlayer: Codable {
    var id: String
    var username: String
    var currency: Int
    var lastUpdate: Date
}


class Player: Identifiable, Codable{
    var id: String
    var username: String
    var pet: Pet?
    var inventory: Inventory?
    var currency: Int
    var lastUpdate: Date
    
    enum CodingKeys: String, CodingKey {
        case id = "user_id"
        case username
        case currency
        case lastUpdate = "last_update"
    }
    
    init(id: String = "test id", username: String = "", pet: Pet? = nil, inventory: Inventory? = nil, currency: Int = 1000, lastUpdate : Date = Date())
    {
        self.id = id
        self.username = username
        self.pet = pet
        self.inventory = inventory
        self.currency = currency
        self.lastUpdate = lastUpdate
    }
}
